File: LeetCode/560SubarraySumEqualsK.py
def subarraySum(nums,k):
    i, res = 0, 0
    for i in range(len(nums)):
        for j in range(i, len(nums)):
            tempSum = sum(nums[i:j+1])
            if tempSum == k:
                res += 1
    return res
    
nums = [1,1,1]
k = 2
result = subarraySum(nums=nums, k=k)
print(result)

File: LeetCode/560SubarraySumEqualsK.java
class SubarraySumEqualsK {
    public static void main(String[] args) {
        int[] arr = {1,1,1};
        //new int[size];
        int k = 1;

        int result = subarraySum(arr,k);
        System.out.println(result);
    }

    private static int subarraySum(int[] arr, int k) {
        int res = 0;
        
        for(int i = 0;i < arr.length;i++) {
            int tempSum = 0;
            for (int j = i; j < arr.length;j++) {
                tempSum += arr[j];
                if (tempSum == k) {
                    res += 1;
                }
            }
        }
        return res;
    }
}